src/macaron/slsa_analyzer/provenance/intoto/v01/__init__.py

```diff
@@ -10,11 +10,29 @@
 from macaron.slsa_analyzer.provenance.intoto.errors import ValidateInTotoPayloadError
 from macaron.util import JsonType
 
-# The list of cryptographic algorithms supported in in-toto v0.1 provenance.
+# The full list of cryptographic algorithms supported in in-toto v0.1 provenance.
 # These are used as keys within the digest set of the resource descriptors within the subject.
-# For the full v0.1 list see: https://github.com/in-toto/attestation/blob/main/spec/v0.1.0/field_types.md#DigestSet
+# For v0.1 see: https://github.com/in-toto/attestation/blob/main/spec/v0.1.0/field_types.md#DigestSet
 VALID_ALGORITHMS = [
+    "sha256",
```

Comment:
Hmm if we have decided to only accept:
- `sha1` for intoto v0.1 DigestSet
- `sha1` and `gitDigest` for intoto v1.0 DigestSet

Why do we still need to include other algorithms as `VALID_ALGORITHM` for both v0.1 and v1.0?
From what I can see:
- The `VALID_ALGORITHMS` variable of `provenance/intoto/v01/__init__.py` is not being used anywhere in the codebase.
- The `VALID_ALGORITHMS` variable of `provenance/intoto/v1/__init__.py` is being used in this method [is_valid_digest_set](https://github.com/oracle/macaron/blob/e36578ee974471f156919bebdfbce5134479982a/src/macaron/slsa_analyzer/provenance/intoto/v1/__init__.py#L176).
- There are some differences between the [intoto/v1/\_\_init\_\_.py::is_valid_digest_set](https://github.com/oracle/macaron/blob/e36578ee974471f156919bebdfbce5134479982a/src/macaron/slsa_analyzer/provenance/intoto/v1/\_\_init\_\_.py#L176) and [intoto/v01/\_\_init\_\_.py::is_valid_digest_set](https://github.com/oracle/macaron/blob/e36578ee974471f156919bebdfbce5134479982a/src/macaron/slsa_analyzer/provenance/intoto/v01/__init__.py#L173).
  - The return type is different.
  - The one for v01 doesn't check the keys in DigestSet against the list of `VALID_ALGORITHMS`.

From my understanding, the usages of `sha1` and/or `gitDigest` are only for obtaining the git commit hash from the DigestSet of provenance. Therefore, they are different from the `VALID_ALGORITHMS` we used to validate the content of DigestSet for the provenance payload. Am I understanding it correctly? If that is True, then I still think we need to agree on whether we should resolve this inconsistencies between the two `is_valid_digest_set` :thinking: 
@nathanwn can you let me know the original reason why the `is_valid_digest_set` doesn't incorporate check for valid algorithms. Was it because we want to accept un-recognized fields as suggested by the intoto payload definition?

Reply:
My understanding:
* The `VALID_ALGORITHMS` sets and the `is_valid_digest_set` functions are used to validate values of the `DigestSet` type (in-toto spec: [v0.1.0](https://github.com/in-toto/attestation/blob/main/spec/v0.1.0/field_types.md), [v1](https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md)). Previously, the `is_valid_digest_set` function for `v01` did not validate the hash algorithms. The new `is_valid_digest_set` function for `v1` now does, though. We probably want to unify these two functions somehow. Not sure if it should be part of this PR. What do you think @benmss?
* The `"sha1"` and `"gitCommit"` keys are only used in the context of extracting commit hashes from provenances, and are not used for validating `DigestSet` values in general.

Reply:
I agree with unifying the two functions. I'll add it to this PR.